### ISSUE TITLE
Problem in proxies

### ISSUE THREAD

**Issue by rui-ribeiro** [flexirest/flexirest#70]
Hi!

I have a problem using the proxy feature. I want to map a response to a array of arrays but when I try that i receive this error: **no implicit conversion of Symbol into Integer**.

I'm using the last version of the gem: 1.3.3.3

I have a code similar to this:

```ruby
class ArticleProxy < Flexirest::ProxyBase
  get "/tags" do
    response = passthrough
    translate(response) do |body|
     body.map! do |article|
        [article['author'], article['name']]
      end
     body
    end
  end
end
``` 

The response is something like: 

```ruby
[ 
     { "id": 1, "author": 'Shakespeare', "name": "Romeu & Julieta" },
     { "id": 2, "author": "Dumas", "name": "3 Mosqueteers" }
]
``` 

I think the problem is in the method **generate_new_object**

More specifically: 

```ruby
..
 if body.is_a? Array
        result = Flexirest::ResultIterator.new(@response)
        body.each do |json_object|
          result << new_object(json_object, @overridden_name)
        end
..
``` 

Can you provide some help? Is this some bug or is not suposed to do that? 

Thanks in advance.

**Comment by andyjeffries** [comment]
When you say you receive that error, can you provide the backtrace for the error? Might help me track down exactly why it's happening. It wasn't intended to handle arrays of arrays of objects, but there's no reason it theoretically shouldn't, so I'd like to get it fixed.

**Comment by rui-ribeiro** [comment]
The backtrace I have is not useful but I can create a demo project, if you want

**Comment by andyjeffries** [comment]
That would be great, thanks.

**Comment by rui-ribeiro** [comment]
Done:

https://github.com/rui-ribeiro/SampleApp

Just run the app, the index action will call User.todos and the proxy will take effect and the error occur.

If you need more information, just ask.

**Comment by rui-ribeiro** [comment]
Hi,

Probably you are very busy, so can I do something to help?

I would submit a PR but I don't know the right way to change this behavior. 

Thanks

**Comment by andyjeffries** [comment]
Sorry mate, I'm launching a new product at work and it's been really busy.

I'll get to this when I can, but I'm snowed-under at the moment.

You said you don't know the right way, if you want to sketch out a solution on here I can then comment if it sounds reasonable/right or not? Then you can try submitting a PR for it?

If not, I'll get to it when I can.

Thanks for your understanding.

**Comment by mikevoets** [comment]
Have you tried to do the following:
```
class ArticleProxy < Flexirest::ProxyBase
  get "/tags" do
    response = passthrough
    translate(response) do |body|
     body.map do |article|
        [article['author'], article['name']]
      end
    end
  end
end
```
Note: I think in general using `map!` is not a good idea since it's slow on large hashes.

**Comment by rui-ribeiro** [comment]
Yes, I tried that but didn't work.

I think Flexirest can't handle arrays of array.

**Comment by andyjeffries** [comment]
Hi Rui, sorry for the HORRENDOUS delay in solving this issue. I've just fixed this issue and released it as v1.6.2. Let me know if it works for you (if you're still using Flexirest).

**Comment by rui-ribeiro** [comment]
Thanks! I will test this later.

Yes, I'm still using flexirest :)
